cpp/src/parquet/chunker_internal_test.cc

```diff
@@ -0,0 +1,1687 @@
+
+    // Assert the third element has length 1
+    ASSERT_TRUE(array->IsValid(i + 2));
+    ASSERT_EQ(list_array->value_length(i + 2), 1);
+
+    // Assert the fourth element has length 3
+    ASSERT_TRUE(array->IsValid(i + 3));
+    ASSERT_EQ(list_array->value_length(i + 3), 3);
+  }
+
+  ASSERT_NOT_OK(GenerateArray(field, 3, /*seed=*/0));
+  ASSERT_OK(GenerateArray(field, 8, /*seed=*/0));
+}
+
+Result<std::shared_ptr<Table>> GenerateTable(
+    const std::shared_ptr<::arrow::Schema>& schema, int64_t size, int64_t seed = 0) {
+  std::vector<std::shared_ptr<Array>> arrays;
+  for (const auto& field : schema->fields()) {
+    ARROW_ASSIGN_OR_RAISE(auto array, GenerateArray(field, size, ++seed));
+    arrays.push_back(array);
+  }
+  return Table::Make(schema, arrays, size);
+}
+
+Result<std::shared_ptr<Table>> ConcatAndCombine(
+    const std::vector<std::shared_ptr<Table>>& parts) {
+  // Concatenate and combine chunks so the table doesn't carry information about
+  // the modification points
+  ARROW_ASSIGN_OR_RAISE(auto table, ConcatenateTables(parts));
+  return table->CombineChunks();
+}
+
+Result<std::shared_ptr<Table>> ReadTableFromBuffer(const std::shared_ptr<Buffer>& data) {
+  std::shared_ptr<Table> result;
+  FileReaderBuilder builder;
+  std::unique_ptr<FileReader> reader;
+  auto props = default_arrow_reader_properties();
+
+  RETURN_NOT_OK(builder.Open(std::make_shared<BufferReader>(data)));
+  RETURN_NOT_OK(builder.memory_pool(::arrow::default_memory_pool())
+                    ->properties(props)
+                    ->Build(&reader));
+  RETURN_NOT_OK(reader->ReadTable(&result));
+  return result;
+}
+
+Result<std::shared_ptr<Buffer>> WriteTableToBuffer(
+    const std::shared_ptr<Table>& table, int64_t min_chunk_size, int64_t max_chunk_size,
+    int64_t row_group_length = 1024 * 1024, bool enable_dictionary = false,
+    ParquetDataPageVersion data_page_version = ParquetDataPageVersion::V1) {
+  auto sink = CreateOutputStream();
+
+  auto builder = WriterProperties::Builder();
+  builder.enable_content_defined_chunking()->content_defined_chunking_options(
+      {min_chunk_size, max_chunk_size, /*norm_level=*/0});
+  builder.data_page_version(data_page_version);
+  if (enable_dictionary) {
+    builder.enable_dictionary();
+  } else {
+    builder.disable_dictionary();
+  }
+  auto write_props = builder.build();
+  auto arrow_props = ArrowWriterProperties::Builder().store_schema()->build();
+  RETURN_NOT_OK(WriteTable(*table, default_memory_pool(), sink, row_group_length,
+                           write_props, arrow_props));
+  ARROW_ASSIGN_OR_RAISE(auto buffer, sink->Finish());
+
+  // check whether the schema has extension types, if not we can easily ensure that
+  // the parquet seralization is roundtripable with CDC enabled
+  bool validate_roundtrip = true;
+  for (const auto& field : table->schema()->fields()) {
+    if (field->type()->id() == ::arrow::Type::EXTENSION) {
+      validate_roundtrip = false;
```

Comment:
Why extension type does not support roundtrip validation? What if the arrow schema has been preserved in the parquet metadata?